app/src/main/java/com/github/multimatum_team/multimatum/activity/CalendarActivity.kt

```diff
@@ -1,58 +1,105 @@
 import com.github.multimatum_team.multimatum.model.Deadline
 import com.github.multimatum_team.multimatum.model.DeadlineState
+import com.github.multimatum_team.multimatum.service.ClockService
 import com.github.multimatum_team.multimatum.util.hideKeyboard
 import com.github.multimatum_team.multimatum.util.hideKeyboardWhenClickingInTheVoid
 import com.github.multimatum_team.multimatum.util.setOnIMEActionDone
 import com.github.multimatum_team.multimatum.viewmodel.DeadlineListViewModel
 import com.google.android.material.textfield.TextInputEditText
 import dagger.hilt.android.AndroidEntryPoint
-import java.time.Instant
 import java.time.LocalDateTime
-import java.time.ZoneId
+import java.util.*
+import javax.inject.Inject
 
 @AndroidEntryPoint
-class CalendarActivity : AppCompatActivity() {
+class CalendarActivity : AppCompatActivity(), EventsCalendar.Callback {
+
+    @Inject
+    lateinit var clockService: ClockService
+
     private val viewModel: DeadlineListViewModel by viewModels()
-    private var selectedDate: LocalDateTime =
-        Instant.now().atZone(ZoneId.systemDefault()).toLocalDateTime()
+    private lateinit var dateSelected: LocalDateTime
 
-    private lateinit var calendarView: CalendarView
+    private lateinit var calendarView: EventsCalendar
     private lateinit var deadlineTitleInputView: TextInputEditText
     private lateinit var addDeadlineButton: Button
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_calendar)
+        dateSelected = clockService.now()
 
         calendarView = findViewById(R.id.calendar_view)
         deadlineTitleInputView = findViewById(R.id.textInputEditCalendar)
         addDeadlineButton = findViewById(R.id.calendar_add_deadline_button)
 
-        initCalendar()
+        // Allow the calendar to go only a month before the actual date and
+        // with no end after it
+        val start = transformLocalDateTimeToCalendar(clockService.now())
+        start.add(Calendar.MONTH, -1)
+        val end = transformLocalDateTimeToCalendar(clockService.now())
+        end.add(Calendar.MONTH, EventsCalendarUtil.DEFAULT_NO_OF_MONTHS / 2)
+
+        // Setup of the calendar
+        calendarView.setSelectionMode(calendarView.SINGLE_SELECTION)
+            .setToday(transformLocalDateTimeToCalendar(dateSelected))
+            .setMonthRange(start, end) //set starting month and ending month
+            .setWeekStartDay(Calendar.MONDAY, false)
+            .setCurrentSelectedDate(transformLocalDateTimeToCalendar(dateSelected))
+            .setEventDotColor(Color.RED)
+            // allow to choose what happen when a date is selected and more
+            // with the 3 override functions
+            .setCallback(this)
+
+        updateColorText()
+        setDotForEveryDeadline()
         initTextInput()
         initAddButton()
     }
 
+    private fun updateColorText(){
```

Comment:
Small formatting issue, please put a space between `)` and `{`.
Sorry I coudln't come up with something more interesting to say, code looks good otherwise